conans/client/settings_preprocessor.py

```diff
@@ -0,0 +1,18 @@
+from conans.errors import InvalidConanSettingField
+
+
+def preprocess(settings, output):
+    fill_runtime(settings, output)
+
+
+def fill_runtime(settings, output):
+    try:
+        if settings.compiler == "Visual Studio":
+            if settings.get_safe("compiler.runtime") is None:
+                settings.compiler.runtime = "MDd" if settings.get_safe("build_type") == "Debug" \
+                                                  else "MD"
+                output.info("Setting 'compiler.runtime' not declared, automatically "
+                            "adjusted to '%s'" % settings.compiler.runtime)
+    except InvalidConanSettingField:  # If the settings structure doesn't match these general
```

Comment:
It's not the same to get "None" from an empty setting (no value in the profile) than: "That setting doesn't exist" (no setting in the setting.yml) and of course we cannot capture any exception not related with that.
We cannot just provide a generic and configurable setting.yml but then expect that the runtime (or the emscripten https://github.com/conan-io/conan/pull/1565) is declared.

The `auto` sounds good, but again, it's not only about the values, you could need to remove a setting in the preprocessing, it's not just a profile thing.